src/components/Axis/HorizontalAxisGridLines.tsx

```diff
@@ -1,35 +1,82 @@
-import { CSSProperties } from 'react';
+import { CSSProperties, useMemo } from 'react';
 
 import { Position } from '../../types';
 
-import { TicksType } from './types';
+import { Scales, TicksType } from './types';
 
 interface HorizontalAxisGridLinesProps {
   plotHeight: number;
   primaryTicks: TicksType[];
   position: Position;
+  scale?: Scales;
   style?: CSSProperties;
+  secondaryGrid?: boolean;
+  secondaryStyle?: CSSProperties;
 }
 
 export default function HorizontalAxisGridLines(
   props: HorizontalAxisGridLinesProps,
 ) {
-  const { plotHeight, style, primaryTicks, position: axisPosition } = props;
-  return (
-    <g>
-      {primaryTicks.map(({ position }) => (
-        <line
-          key={position}
-          x1={position}
-          x2={position}
-          y1={axisPosition === 'top' ? plotHeight : -plotHeight}
-          y2="0"
-          stroke="black"
-          strokeDasharray="2,2"
-          strokeOpacity={0.5}
-          style={style}
-        />
-      ))}
-    </g>
-  );
+  const {
+    plotHeight,
+    style,
+    primaryTicks,
+    position: axisPosition,
+    secondaryGrid,
+    secondaryStyle,
+    scale,
+  } = props;
+  const Grid = useMemo(() => {
+    const Grid = primaryTicks.map(({ position }) => (
+      <line
+        key={position}
+        x1={position}
+        x2={position}
+        y1={axisPosition === 'top' ? plotHeight : -plotHeight}
+        y2="0"
+        stroke="black"
+        strokeDasharray="2,2"
+        strokeOpacity={0.5}
+        style={style}
+      />
+    ));
+
+    if (secondaryGrid) {
+      const density = 5;
+      const secondaryTicks = scale?.ticks(primaryTicks.length * density) || [];
+      secondaryTicks.forEach((tick) => {
+        const position = scale?.(tick);
+        if (
```

Comment:
don't we already compute the secondary ticks somewhere (to display them)? We could move that higher in the tree to avoid computing them in multiple places.

Reply:
this is not secondaryTicks, just need to rename the variable
also secondaryTicks and secondaryGrid are not the same always

Reply:
we can add `secondaryTicksDensity` , `secondaryGridDensity` props in `Axis`